src/lib/cinchy.service.ts

```diff
@@ -68,29 +93,64 @@ export class CinchyService {
 
         let that = this;
         let emitInfo = true;
+
         if (this._oAuthService.hasValidAccessToken()) {
             emitInfo = false;
         }
-        return new Promise<boolean>(function(resolve, reject) {
+
+        // In the event that there is a querystring present in the URL, it is likely to be lost when the login flow
+        // redirects back to the application. To prevent that, we save it here and then re-add it after the app returns
+        if (location.search.length) {
+            this._oAuthStorage.setItem("[Cinchy][login][queryParams]", location.search.substr(1));
+        }
+
+        return new Promise<boolean>(function (resolve, reject) {
             that._oAuthService.loadDiscoveryDocumentAndLogin()
-            .then(response => {
-                resolve(response);
-                if (response) {
-                    if (emitInfo) {
-                        that.emitAccessToken();
-                        that.emitIdentityClaims();
+                .then((response: boolean) => {
```

Comment:
One last thing to note, not sure if I fully understand when the response is false, but we may want to test when the response is false, if there is a need for the routerSubscription because it may not redirect in that scenario